src/models/Links.php

```diff
@@ -93,6 +94,74 @@ public function readAll(): array
     }
 
     /**
+     * Get related entities
+     *
+     * @param string $type Type of related entities (items or experiments)
+     * @return array
+     */
+    public function readRelated(string $type): array
+    {
+        if (!($type === 'items' or $type === 'experiments')) {
+            throw new ImproperActionException(_('Internal error.'));
```

Comment:
Use the `Tools::error()` function instead. Also, this would create yet another string to translate which is something to avoid.

Reply:
ImproperAction is for something that the user might do but it's not allowed so we kindly give a meaningful error message, as ImproperAction message is shown to the user (wheras others, a generic message is shown and the full error is logged)

Reply:
I changed it to an Exception that will be caught in web/database.php. It will result in an error message from `Tools::error()`.
However, this is just to prevent a call of `readRelated($type)` without a proper parameter. There is no user input and the problem should ~~never happen~~ only happen during development. Maybe the if statement can be removed.

Reply:
IMHO this check should be removed. I was also thinking that because this function will always be called with experiments AND items, remove the argument and make it return an array that contains an array for experiments and one for items. Just a suggestion.

Reply:
I like your suggestion.
It is implemented now. However, it uses associative array destructuring which is only available since PHP 7.1. Is this a problem?